control-plane/pkg/reconciler/consumergroup/controller.go

```diff
@@ -202,7 +202,32 @@ func ResyncOnStatefulSetChange(ctx context.Context, handle func(interface{})) {
 			ss := obj.(*appsv1.StatefulSet)
 			return ss.GetNamespace() == systemNamespace && kafkainternals.IsKnownStatefulSet(ss.GetName())
 		},
-		Handler: controller.HandleAll(handle),
+		Handler: cache.ResourceEventHandlerFuncs{
+			AddFunc: handle,
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				o, ok := oldObj.(*appsv1.StatefulSet)
+				if !ok {
+					return
+				}
+				n, ok := newObj.(*appsv1.StatefulSet)
+				if !ok {
+					return
+				}
+
+				// This should never happen, but we check for nil to be sure.
+				if o.Spec.Replicas == nil || n.Spec.Replicas == nil {
+					return
+				}
+
+				// Only handle when replicas change
+				if *o.Spec.Replicas == *n.Spec.Replicas && o.Status.ReadyReplicas == n.Status.ReadyReplicas {
```

Comment:
I think we currently only look at `Spec.Replicas` when scheduling, maybe we can simplify this check even further?